### PR TITLE
Mind Control Refactoring

### PR DESCRIPTION
**Pull request by zumbak04**
## Secret Changelog:
- Refactored the script of Mind Control events. Improvements:
  - Created 3 new effects: `remove_mind_control_modifier_effect`, `increase_mind_control_stage_effect`, `decrease_mind_control_stage_effect`.
  - Created 1 new trigger: `can_be_mind_controlled_by_scourge_trigger`
  - The event that increases your **Mind Control Modifier** (`WCUND.1106`) and the event that decrease your **Mind Control Modifier** (`WCUND.1107`) can't create a few event windows. For example, when you decrease **Mind Control Variable** by using `event WCUND.1112` console command, the event shall fire at 75 **Mind Control Variable**. If you continue decreasing **Mind Control Variable** without closing the event window, the game shall not open another event window.
- Refactored WCUND.1109. Nothing special, just used `else_if` and `else` and replaced `is_ruler` by `is_landed` because ruler may be unlanded.
- Renamed `mind_control_2` because `mind_control_2` and `mind_control_3` had the same name.

## How to test
Start as the Lich King in the 603 bookmark, use `event WCUND.1111` and `event WCUND.1112` to increase/decrease **Mind Control Variable** a lot of times and see what happens. Try to break it somehow.

### CODE REVIEW

File: common/scripted_triggers/wc_scripted_triggers.txt [hunk 1]
```diff
@@ -4368,4 +4368,11 @@ not_raising_undead_flag_trigger = {
 		text = raising_undead_flag_trigger_tooltip
 		NOT = { has_character_flag = raising_undead_flag }
 	}
+}
+
+# Trigger used in mind control events
+can_be_mind_controlled_by_scourge_effect = {
```
Comment (MemeWizard) [inline common/scripted_triggers/wc_scripted_triggers.txt]: trigger

Comment (zumbak04) [reply common/scripted_triggers/wc_scripted_triggers.txt]: Renamed.

**Comment by MemeWizard** [comment]
When you increase MC power from value below 75 to value above 75, MC powers sets to 100.

<details>
<summary>Click to expand</summary>

image
image
image

</details>

**Comment by zumbak04** [comment]
> When you increase MC power from value below 25/50/75 to value above 25/50/75, MC powers sets to 100.

Fixed?!

File: localisation/00_zuwc_united_localisation.csv [hunk 2]
```diff
@@ -3024,6 +3020,10 @@ RULE_ADOPTION_NO_PLAYER_DYNASTY;Restricted;;;;;;;;;;;;;x
 RULE_ADOPTION_NO_PLAYER_DYNASTY_DESC;Only players can adopt newly-spawned characters by adoption event into the player's dynasty.;;;;;;;;;;;;;x
 RULE_ADOPTION_DISABLED;Disabled;;;;;;;;;;;;;x
 RULE_ADOPTION_DISABLED_DESC;Adoption event is disabled.;;;;;;;;;;;;;x
+mind_control_1;Strong Mind Control;Contr�le mental fort;Starke Gedankenkontrolle;;;;;;;;;;;x
+mind_control_2;Weaking Mind Control;;;;;;;;;;;;;x
```
Comment (MemeWizard) [inline localisation/00_zuwc_united_localisation.csv]: Maybe weakening?
Also, this choice is weird, it implies that mind control power is being weakening, but what if it actually growing stronger?

File: localisation/00_zuwc_united_localisation.csv [hunk 3]
```diff
@@ -3024,6 +3020,10 @@ RULE_ADOPTION_NO_PLAYER_DYNASTY;Restricted;;;;;;;;;;;;;x
 RULE_ADOPTION_NO_PLAYER_DYNASTY_DESC;Only players can adopt newly-spawned characters by adoption event into the player's dynasty.;;;;;;;;;;;;;x
 RULE_ADOPTION_DISABLED;Disabled;;;;;;;;;;;;;x
 RULE_ADOPTION_DISABLED_DESC;Adoption event is disabled.;;;;;;;;;;;;;x
+mind_control_1;Strong Mind Control;Contr�le mental fort;Starke Gedankenkontrolle;;;;;;;;;;;x
+mind_control_2;Weakened Mind Control;;;;;;;;;;;;;x
```
Comment (DukeJeans) [inline localisation/00_zuwc_united_localisation.csv]:
```suggestion
mind_control_2;Steady Mind Control;;;;;;;;;;;;;x
```

Comment (zumbak04) [reply localisation/00_zuwc_united_localisation.csv]: Done!

**Comment by zumbak04** [comment]
> Cheat raping the Lich King was great fun, would do again. 10/10

I hope you will.

**Pull request merged**